mpv-full
[PATCH] fix build when using ffmpeg git master

Backport of:
https://github.com/mpv-player/mpv/commit/1b19ba6c57c9adabed2bbb22b6c0c231c44d0af4

Related links:
https://github.com/mpv-player/mpv/issues/5813
https://github.com/mpv-player/mpv/pull/5814

diff --git a/waftools/fragments/cuda.c b/waftools/fragments/cuda.c
deleted file mode 100644
index 1d534f62d3..0000000000
--- a/waftools/fragments/cuda.c
+++ /dev/null
@@ -1,15 +0,0 @@
-#define CUDA_VERSION 7050
-
-typedef void * CUcontext;
-
-#include <libavcodec/avcodec.h>
-#include <libavutil/hwcontext.h>
-#include <libavutil/hwcontext_cuda.h>
-
-int main(int argc, char *argv[]) {
-    enum AVHWDeviceType type = AV_HWDEVICE_TYPE_CUDA;
-    AVCUDADeviceContextInternal *foo;
-    AVCodecContext *avctx = avcodec_alloc_context3(NULL);
-    avctx->hw_device_ctx = NULL;
-    return 0;
-}
diff -Naurp a/wscript b/wscript
--- a/wscript	2018-02-13 02:07:05.000000000 +0000
+++ b/wscript	2018-05-11 20:14:33.387027194 +0000
@@ -850,8 +850,7 @@ hwaccel_features = [
         'name': '--cuda-hwaccel',
         'desc': 'CUDA hwaccel',
         'deps': 'gl',
-        'func': check_cc(fragment=load_fragment('cuda.c'),
-                         use='libavcodec'),
+        'func': check_true,
     }
 ]
 
